WordPress/Classes/Stores/RemoteFeatureFlagStore.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import WordPressKit
+
+class RemoteFeatureFlagStore {
+
+    init() {
+        DDLogInfo("🚩 Remote Feature Flag Device ID: \(deviceID)")
+    }
+
+    /**
+     Fetches remote feature flags from the server.
+
+     - Parameter callback: An optional callback that can be used to update UI following the fetch. It is not called on the UI thread.
+    */
+    public func update(then callback: FetchCallback? = nil) {
+        // We currently use an anonymous connection to WordPress.com, because we don't need user state.
+        // In the future, we could provide login credentials to the endpoint, which would allow customizing flags server-side on a per-user basis.
+        let remote = FeatureFlagRemote(wordPressComRestApi: WordPressComRestApi.defaultApi())
+        remote.getRemoteFeatureFlags(forDeviceId: deviceID) { [weak self] result in
+            switch result {
+                case .success(let flags):
+                    self?.cache = flags.dictionaryValue
+                    DDLogInfo("🚩 Successfully updated local feature flags: \(flags)")
+                    callback?()
+                case .failure(let error):
+                    DDLogError("🚩 Unable to update Feature Flag Store: \(error.localizedDescription)")
+            }
+        }
+    }
+
+    /// Checks if the local cache has a value for a given `FeatureFlag`
+    public func hasRemoteValueForFlag(_ flag: FeatureFlag) -> Bool {
+        return flag.remoteKey != nil && cache[flag.remoteKey!] != nil
+    }
+
+    /// Looks up the value for a remote feature flag.
+    ///
+    /// If the flag exists in the local cache, the current value will be returned.  If the flag does not exist in the local cache, the compile-time default will be returned.
+    /// - Parameters:
+    ///     - flag: The `FeatureFlag` object associated with a remote feature flag
+    public func valueForFlag(_ flag: FeatureFlag) -> Bool {
+        guard
+            let remoteKey = flag.remoteKey, // Not all flags need remote keys, since they may not use remote feature flagging
```

Comment:
IMHO this comment here should be part of the documentation in `FeatureFlag.remoteKey` instead of being present here, since that's intrinsic to `remoteKey` and not really to this particular guard.

Reply:
I ended up putting it in both places in dd84a9bd8702897382058007f94d4ace1c393c8b. 

I agree that it's important to have it in the protocol definition for sure, but this _should_ be the only place in the codebase that this value is examined, and it may not be immediately clear _why_ it needs to be unwrapped when reading the code, which is why I left the comment in there.

Hope that's ok?